.github/workflows/docs-markdownlint.yml

```diff
@@ -0,0 +1,34 @@
+name: Documentation Markdown Lint
+
+# This workflow runs MarkDownLint on changed docs files in pull requests. More info: https://docs.airbyte.com/platform/next/contributing-to-airbyte/writing-docs#lint-with-markdownlint
+
+on:
+  pull_request:
+    # Only trigger if Markdown files in /docs/ change
+    paths:
+      - "docs/**/*.md"
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the repository
+      - uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+
+      # Detect changed Markdown files in the PR
+      - uses: tj-actions/changed-files@v46
```

Comment:
One comment on this is that we should pin this to a specific commit as this specific action has been compromised in the past; https://www.stepsecurity.io/blog/harden-runner-detection-tj-actions-changed-files-action-is-compromised

Probably a good idea for us to do this for all actions in general, but this one in particular is probably important to pin.